src/modules/csm/checkpoint.py

```diff
@@ -3,25 +3,28 @@
 from dataclasses import dataclass
 from itertools import batched
 from threading import Lock
-from typing import Iterable, Sequence
+from typing import Iterable, Sequence, TypeGuard
 
 from src import variables
 from src.constants import SLOTS_PER_HISTORICAL_ROOT
-from src.metrics.prometheus.csm import CSM_UNPROCESSED_EPOCHS_COUNT, CSM_MIN_UNPROCESSED_EPOCH
+from src.metrics.prometheus.csm import CSM_MIN_UNPROCESSED_EPOCH, CSM_UNPROCESSED_EPOCHS_COUNT
 from src.modules.csm.state import State
 from src.providers.consensus.client import ConsensusClient
-from src.providers.consensus.types import BlockAttestation
+from src.providers.consensus.types import BlockAttestationElectra, BlockAttestationPhase0
 from src.types import BlockRoot, BlockStamp, EpochNumber, SlotNumber, ValidatorIndex
 from src.utils.range import sequence
 from src.utils.timeit import timeit
+from src.utils.types import hex_str_to_bytes
 from src.utils.web3converter import Web3Converter
 
 logger = logging.getLogger(__name__)
 lock = Lock()
 
 
-class MinStepIsNotReached(Exception):
-    ...
+class MinStepIsNotReached(Exception): ...
+
+
+type BlockAttestation = BlockAttestationPhase0 | BlockAttestationElectra
```

Comment:
We have already declaration of BlockAttestation in another file.

Reply:
I've changed it a little bit, so it's gone now.